Верстка userBlocK(svg)

Elizaveta810 opened the pull request:
Дописала стили по userBlocK. Подтянула svg(отображаются корректно)

Elizaveta810 committed:
Верстка userBlocK(svg)

Elizaveta810 committed:
логика работы кнопки войти

aloygolone commented:
Отлично, частично поправила то, что писал в телеграмм. Осталось разобраться с:
1. функция handleSigninForm управляет вызовом формы SigninForm (можно ее переименовать в handleOpenSigninForm - так будет понятнее, зачем нужна эта функция) - то есть она должна менять состояние, например isOpenedSigninForm на true при нажатии кнопки "Войти" в хедере (то есть состояние "авторизованного пользователя" здесь не причем)
2. Значит внутри верстки (после return) - внизу ты можешь добавить компонента SigninForm и прописать логику его появления, которая завязана на том, что я описал в 1 пункте.
3. Также внутри верстки внизу для компонента <UserModal /> можешь прописать логику отображения этого компонента

Elizaveta810 committed:
Доработала замечания, по логике

aloygolone commented:
Продолжаем правку (делай прямо по пунктам):
1. 8 строчка - давай назовем состояние [isOpenedSigninForm, setIsOpenedSigninForm] - чтобы было однотипно с остальными (не забываем вносить правки в код в связи с изменениями)
2. После 9 строчки добавляем новое состояние для открытия UserModal - [isOpenedUserModal, setIsOpenedUserModal]
3. Также добавляем состояние в соответствии с "программой разработки приложения" (лист "состояния компонентов") для [currentPage, setCurrentPage] - и прописываем логику работы состояния
4. 22-24 строчки можно разкомментировать - нам нужна эта функция для открытия UserModal
5. с 42 по 58 строчку отображения компонентов можно сократить, используя тернарный оператор
{isAuthorizated ? (юзерблок) : (кнопка Войти)}
6. 60 строчка - внутрь компонента SigninForm прокидываем пропс setIsOpenedSigninForm - при прохождении авторизации функция должна стать false (или при замене на окно регистрации) - чтобы форма закрывалась
7. 61 строчка - возвращаем логику отображения UserModal - {isOpenedUserModal && <UserModal />}
8. Не забываем про импорты всех сторонних компонентов (должны подтянуться)

Elizaveta810 committed:
Доработала логику components_Header_UserModal

Elizaveta810 commented:
Все поправки по рекомендациям выше внесены. Также в components UserModal, обработала кнопку Выйти (ПРОВЕРЬ ПОЖАЛУЙСТА ПРАВИЛЬНО ЛИ Я ЭТО СДЕЛАЛА).

aloygolone committed:
Update Header.tsx

Внёс корректировку: на лого вместо тега <a> поставил <Link> с ссылкой на главную

aloygolone commented:
Принимаю, остальные правки будем вносить при проверки работы приложения
